### PR TITLE
Runtimes workspace dependencies & cleaning

### LINKED ISSUE #1582: Reexport common dependencies from runtime-common
### Problem
For each pallet/runtime dependency used, we need to add such dependency to:
- `development-runtime`
- `altair-runtime`
- `centrifuge-runtime`
- `integration-tests`
- Sometimes to `runtime-common`

And configure `runtime-benchmarks` and `try-runtime` features in all crates. 

Avoiding/mixing some feature names will lead to an annoying compile error that could take time to figure out what's going on.

### Solution
We can simplify all these dependency boilerplate by adding all pallets to `runtime-common` and allowing `runtime-common` to reexport them. So, in `runtime-common` we can have:

```rust
pub mod reexport {
    pub mod pallets {
        pub use pallet_timestamp;
        // Other pallets
    }
}
```

From runtimes, on top of the file:
```rust
use runtime_common::reexport::pallets::*;
```

And the code would remain inmutable for runtimes, but with access to all pallets without modifying `Cargo.toml` files.

In the future, this would avoid `nix` hash breaks.

### PR DESCRIPTION
**Pull request by lemunozm**
# Description

Refactorizations and cleanings in the runtime part.

The first step to get this: https://github.com/centrifuge/centrifuge-chain/issues/1380

Fixes ~~#1582~~

## Changes and Descriptions
- [x] Reorganize the fees section in `runtime-common`
- [x] Remove `scripts/runtime-weight-template.hbs` because now we use the default from substrate.
- [x] ~~`runtime-common` export a list with all pallets used in our runtimes, to avoid importing them if all of our runtimes.~~
  - There is an issue with the autogenerated weights where the default template is not valid if we reimport the pallets, we would need to maintain a template by us (the one we currently have is not used). **Do we want to maintain the template to get this reexporting list?**
    - Finally, it seems more painful to have an exporting list than to copy the pallet list.
- [x] Using workspace dependencies in `runtime-common`
- [x] Using workspace dependencies in `development`
- [x] Using workspace dependencies in `altair`
- [x] Using workspace dependencies in `centrifuge`
- [x] Using workspace dependencies in `integration-tests`
- [ ] ~~Factorize types from `development` into `runtime-common`~~ => Other PR
- [ ] ~~Factorize types from `altair` into `runtime-common`~~=> Other PR
- [ ] ~~Factorize types from `centrifuge` into `runtime-common`~~ => Other PR

### CODE REVIEW

**Comment by lemunozm** [comment]
Currently `cargo` complains (with a warning) about workspace inheritance with renaming packages (see the [issue](https://github.com/rust-lang/cargo/issues/12546)). We can:
- Life with the warning until it be (hopefully) solved.
- Change `codec` to `parity_scale_codec` everywhere in our database

File: Cargo.toml [hunk 1]
```diff
@@ -158,7 +352,7 @@ runtime-common = { path = "runtime/common" }
 frame-benchmarking = { git = "https://github.com/paritytech/substrate", optional = true, default-features = false, branch = "polkadot-v0.9.43" }
 frame-system-benchmarking = { git = "https://github.com/paritytech/substrate", optional = true, default-features = false, branch = "polkadot-v0.9.43" }
 # integration testing
-runtime-integration-tests = { path = "runtime/integration-tests", optional = true, default-features = false }
+#runtime-integration-tests = { path = "runtime/integration-tests", optional = true, default-features = false }
```
Comment (mustermeiszer) [inline Cargo.toml:355]: ???

Comment (wischli) [inline Cargo.toml:355]: Artifact?

File: Cargo.toml [hunk 2]
```diff
@@ -222,7 +416,7 @@ std = [
   "pallet-pool-system/std",
   "polkadot-primitives/std",
   "runtime-common/std",
-  "runtime-integration-tests/std",
+  #"runtime-integration-tests/std",
```
Comment (wischli) [inline Cargo.toml:419]: Artifact? Plus the other commented out features of `runtime-integration-tests` below

File: runtime/integration-tests/src/utils/extrinsics.rs [hunk 3]
```diff
@@ -151,8 +151,6 @@ fn signed_extra_relay(nonce: RelayIndex) -> RelaySignedExtra {
 		frame_system::CheckNonce::<RelayRuntime>::from(nonce),
 		frame_system::CheckWeight::<RelayRuntime>::new(),
 		pallet_transaction_payment::ChargeTransactionPayment::<RelayRuntime>::from(0),
-		#[cfg(not(feature = "runtime-development"))]
-		polkadot_runtime_common::claims::PrevalidateAttests::<RelayRuntime>::new(),
```
Comment (mustermeiszer) [inline runtime/integration-tests/src/utils/extrinsics.rs:155]: Why the removal?

File: runtime/altair/Cargo.toml [hunk 4]
```diff
@@ -1,179 +1,213 @@
...
-  "cumulus-pallet-session-benchmarking/std",
-  "cumulus-pallet-xcm/std",
-  "cumulus-pallet-xcmp-queue/std",
+  "getrandom/std",
+  "hex/std",
+  "scale-info/std",
+  "serde/std",
+  "log/std",
+
+  # Substrate related
```
Comment (wischli) [inline runtime/altair/Cargo.toml:156]: 🚀

File: runtime/altair/Cargo.toml [hunk 5]
```diff
@@ -382,29 +425,28 @@ try-runtime = [
...
-  "runtime-common/try-runtime",
-  "sp-runtime/try-runtime",
 ]
 
 # A feature that should be enabled when the runtime should be build for on-chain
 # deployment. This will disable stuff that shouldn't be part of the on-chain wasm
 # to make it smaller like logging for example.
 on-chain-release-build = [
   "sp-api/disable-logging",
+  "runtime-common/on-chain-release-build",
```
Comment (wischli) [inline runtime/altair/Cargo.toml:449]: Can you give some details about this addition?

File: scripts/runtime-weight-template.hbs [hunk 6]
```diff
@@ -1,47 +0,0 @@
-//! Autogenerated weights for {{pallet}}
-//!
-//! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION {{version}}
-//! DATE: {{date}}, STEPS: `{{cmd.steps}}`, REPEAT: {{cmd.repeat}}, LOW RANGE: `{{cmd.lowest_range_values}}`, HIGH RANGE: `{{cmd.highest_range_values}}`
-//! EXECUTION: {{cmd.execution}}, WASM-EXECUTION: {{cmd.wasm_execution}}, CHAIN: {{cmd.chain}}, DB CACHE: {{cmd.db_cache}}
```
Comment (wischli) [inline scripts/runtime-weight-template.hbs:5]: Why can we remove the bench template?

**Review by wischli — COMMENTED** [review]
Thanks for cleaning up and making our future lives much easier! I have a few questions, overall LGTM.

File: Cargo.toml (hunk 1 above)
Comment (lemunozm) [reply Cargo.toml:355]: Why were we linking integration tests as a dependency of our node before?

File: runtime/integration-tests/src/utils/extrinsics.rs (hunk 3 above)
Comment (lemunozm) [reply runtime/integration-tests/src/utils/extrinsics.rs:155]: Nongeneric tests are always run in CI under `runtime-development,` so this implies that that line is never compiled. In the case of other runtimes, we have now the generic part which should take care of that. Actually, we no longer have `runtime-x` features.

Please correct me if I'm missing something here, but CI will always tests against the development.

File: Cargo.toml [hunk 7]
```diff
@@ -70,9 +70,203 @@ members = [
   "runtime/integration-tests",
 ]
 
+[workspace.package]
+authors = ["Centrifuge <admin@centrifuge.io>"]
+edition = "2021"
+license = "LGPL-3.0"
+homepage = "https://centrifuge.io/"
+repository = "https://github.com/centrifuge/centrifuge-chain"
+documentation = "https://github.com/centrifuge/centrifuge-chain"
```
[the remark this answers is not in the document]
Comment (lemunozm) [reply Cargo.toml:79]: Right! I thought I copied the correct link 🤦🏻‍♂️

File: Cargo.toml (hunk 1 above)
Comment (lemunozm) [reply Cargo.toml:355]: Answer above to Frederik, we do not want integration tests as a dependency of our centrifuge node, right?

File: Cargo.toml (hunk 2 above)
Comment (lemunozm) [reply Cargo.toml:419]: related to the previous answer

File: runtime/altair/Cargo.toml (hunk 5 above)
Comment (lemunozm) [reply runtime/altair/Cargo.toml:449]: Yes, I understand that if we compile altair with `on-chain-release-build` we would want the runtime common used to have that related feature also.

File: scripts/runtime-weight-template.hbs (hunk 6 above)
Comment (lemunozm) [reply scripts/runtime-weight-template.hbs:5]: We're not using it, if we do not put any template we're using the default. That it's how it's working right now.

File: Cargo.toml (hunk 7 above)
[the remark this answers is not in the document]
Comment (lemunozm) [reply Cargo.toml]: Done!

File: Cargo.toml (hunk 1 above)
Comment (mustermeiszer) [reply Cargo.toml:355]: I thought we needed it for some features. Fine then

Comment (NunoAlexandre) [reply Cargo.toml:355]: @lemunozm yeah good point but then we should remove these lines all together.

**Review by mustermeiszer — APPROVED** [review]

**Pull request merged**